dldashboard/web/design_doc.md

```diff
@@ -0,0 +1,41 @@
+A high level design diagram of the current architecture of the dashboard is :
+
+![dashboard architecture diagram](https://craftassist.s3-us-west-2.amazonaws.com/pubr/backend+architecture.png)
+
+In the diagram above:
+- frontend: The React based frontend that hosts other React subcomponents.
+- StateManager: This module manages messaging between frontend and backend. We set the state of the frontend components through the data received socket.io connections between stateManager and backend. 
+- backend : This is the agent process that sends and receives data from frontend. The agent process and its modules (perception, memory, mover, controller etc) can communicate using the the websocket transport.
```

Comment:
how does the backend receive data from the frontend? There are no arrows in the diagram from the frontend to the backend to suggest that.

Reply:
Through the stateManager there is a double sided arrow from StateManager to backend.
Backend communicates to statemanager which communicates the state to frontend and vice versa